cmd/oras/root/push.go

```diff
@@ -108,6 +108,16 @@ Example - Push file "hi.txt" into an OCI image layout folder 'layout-dir' with t
 			if err := option.Parse(cmd, &opts); err != nil {
 				return err
 			}
+
+			if opts.manifestConfigRef != "" && opts.artifactType == "" {
+				switch opts.Flag {
+				case option.ImageSpecAuto:
+					opts.PackVersion = oras.PackManifestVersion1_0
```

Comment:
Better explain why we need to switch here, e.g. here packs an artifact that is not compliant to OCI artifact guidance https://github.com/opencontainers/image-spec/blob/e7f7c0ca69b21688c3cea7c87a04e4503e6099e2/manifest.md?plain=1#L170

Reply:
added a comment